File: Dec2020/30Dec/1-bitAnd2-bitCharacters.cpp
/*
Problem name: 717. 1-bit and 2-bit Characters
Problem link: https://leetcode.com/problems/1-bit-and-2-bit-characters/
*/

class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        for(int i=0;i<bits.size();++i){
            if(i == bits.size()-1)
                return true;
            if(bits[i] == 1)
                i++;
        }
        return false;
    }
};

File: Dec2020/30Dec/1-bitAnd2-bitCharacters.java
/*
Problem Name: 717. 1-bit and 2-bit Characters
Problem Link: https://leetcode.com/problems/1-bit-and-2-bit-characters/
*/
class Solution {
    public boolean isOneBitCharacter(int[] bits) {
        int i;
        for(i=0;i<bits.length - 1;i++)
            if(bits[i] == 1)
                i++;
        return bits.length - 1 == i;
    }
}
